Scenario3/PetImagesTest/StorageMocks/MockBlobContainer.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Concurrent;
+using System.Threading.Tasks;
+using PetImages.Storage;
+
+namespace PetImagesTest.StorageMocks
+{
+    // Buggy mock, need to fix these mocks as part of third scenario.
+    internal class MockBlobContainerProvider : IBlobContainer
+    {
+        private readonly ConcurrentDictionary<string, ConcurrentDictionary<string, byte[]>> Containers;
+
+        internal MockBlobContainerProvider()
+        {
+            this.Containers = new ConcurrentDictionary<string, ConcurrentDictionary<string, byte[]>>();
+        }
+
+        public Task CreateContainerAsync(string containerName)
+        {
+            return Task.FromResult(this.Containers.TryAdd(containerName, new ConcurrentDictionary<string, byte[]>()));
```

Comment:
@imnaseer Need to verify if this mocks are faulty or not. Changed it from the correct mocks to a faulty version I know.